business/definitionService.js

```diff
@@ -82,10 +82,21 @@ class DefinitionService {
     const cached = await this.cache.get(cacheKey)
     if (cached) return cached
     const stored = await this.definitionStore.get(coordinates)
-    if (stored) await this.cache.set(cacheKey, stored)
+    if (stored) this._setDefinitionInCache(cacheKey, stored)
     return stored
   }
 
+  async _setDefinitionInCache(cacheKey, itemToStore) {
+    // 1000 is a magic number here -- we don't want to cache very large definitions, as it can impact redis ops
+    if (itemToStore.files && itemToStore.files.length > 1000) {
```

Comment:
Can itemToStore be null/undefined?

Reply:
Nope, both callers protect against that (although one caller would just blow up before hand).